examples/material_gallery/lib/demo/colors_demo.dart

```diff
@@ -0,0 +1,133 @@
+  const ColorSwatch(name: 'Orange', colors: Colors.orange, accentColors: Colors.orangeAccent, threshold: 700),
+  const ColorSwatch(name: 'Deep Orange', colors: Colors.deepOrange, accentColors: Colors.deepOrangeAccent, threshold: 400),
+  const ColorSwatch(name: 'Brown', colors: Colors.brown, threshold: 200),
+  const ColorSwatch(name: 'Grey', colors: Colors.grey, threshold: 500),
+  const ColorSwatch(name: 'Blue Grey', colors: Colors.blueGrey, threshold: 500)
+];
+
+
+class ColorItem extends StatelessComponent {
+  ColorItem({ Key key, this.index, this.color, this.prefix: "" }) : super(key: key) {
+    assert(index != null);
+    assert(color != null);
+    assert(prefix != null);
+  }
+
+  final int index;
+  final Color color;
+  final String prefix;
+
+  String colorString() => "#${color.value.toRadixString(16).padLeft(8, '0').toUpperCase()}";
+
+  Widget build(BuildContext context) {
+    return new Container(
+      height: kColorItemHeight,
+      padding: const EdgeDims.symmetric(horizontal: 16.0),
+      decoration: new BoxDecoration(backgroundColor: color),
+      child: new Row(
+        justifyContent: FlexJustifyContent.spaceBetween,
+        alignItems: FlexAlignItems.center,
+        children: <Widget> [
+          new Text('${prefix}${index}'),
+          new Text(colorString())
+        ]
+      )
+    );
+  }
+}
+
+class ColorSwatchTabView extends StatelessComponent {
+  ColorSwatchTabView({ Key key, this.swatch }) : super(key: key) {
+    assert(swatch != null && swatch.isValid);
+  }
+
+  final ColorSwatch swatch;
+  final blackTextStyle = Typography.black.body1;
+  final whiteTextStyle = Typography.white.body1;
+
+  Widget build(BuildContext context) {
+    List<Widget> colorItems =  swatch.colors.keys.map((int index) {
+      return new DefaultTextStyle(
+        style: index > swatch.threshold ? whiteTextStyle : blackTextStyle,
+        child: new ColorItem(index: index, color: swatch.colors[index])
+      );
+    })
+    .toList();
+
+    if (swatch.accentColors != null) {
+      colorItems.addAll(swatch.accentColors.keys.map((int index) {
+        return new DefaultTextStyle(
+          style: index > swatch.threshold ? whiteTextStyle : blackTextStyle,
+          child: new ColorItem(index: index, color: swatch.accentColors[index], prefix: 'A')
+        );
+      })
+      .toList());
+    }
+
+    return new ScrollableList(
+      itemExtent: kColorItemHeight,
+      children: colorItems
+    );
+  }
+}
+
+class ColorsDemo extends StatelessComponent {
+  Widget build(BuildContext context) {
+    return new TabBarSelection<ColorSwatch>(
+      values: colorSwatches,
+      child: new Scaffold(
+        toolBar: new ToolBar(
+          elevation: 0,
+          center: new Text("Colors"),
+          tabBar: new TabBar<ColorSwatch>(
+            isScrollable: true,
+            labels: new Map.fromIterable(colorSwatches, value: (ColorSwatch swatch) {
+              return new TabLabel(text: swatch.name.toUpperCase());
```

Comment:
toUpperCase() is problematic for l10n. Can we hard-code the uppercase strings in the data table instead?